docs/fsharp-cheatsheet.md

```diff
@@ -555,136 +555,158 @@ Another way of implementing interfaces is to use *object expressions*.
+| Discriminated Union                                             | `type internal MyDiscUni = A \| B`         |
+| Discriminated Union [ctor](#CodeOrganization_SmartConstructors) | `type MyDiscUni = private A \| B `         |
+| Class                                                           | `type internal MyClass() =`                |
+| Class [ctor](#CodeOrganization_SmartConstructors)               | `type MyClass private () =`                |
+| Class Additional [ctor](#CodeOrganization_SmartConstructors)    | `internal new() = MyClass("defaultValue")` |
+| Class .. `let`                                                  | *Always private. Cannot be modified*       |
+| `type` .. `member`                                              | `member private _.typeMember =`            |
+| `type` .. `val`                                                 | `val internal explicitInt : int`           |
+
+<a name="CodeOrganization_SmartConstructors"></a>
+##### Smart Constructors
+
+Making a primary constructor (ctor) `private` or `internal` is a common convention for ensuring value integrity;
+otherwise known as ["making illegal states unrepresentable" (YouTube:Effective ML)](https://youtu.be/-J8YyfrSwTk?si=ml3AWro6jG77F0YW&t=1080).
+
+Example of Single-case Discriminated Union with a `private` constructor that constrains a quantity between 0 and 100:
 
     type UnitQuantity =
         private UnitQuantity of int
-        with
-        static member private MaxQty = 100
-        static member Create (qty:int) : Result<UnitQuantity, string> =
-            if qty <= UnitQuantity.MaxQty
-            then Ok (UnitQuantity qty)
-            else Error $"UnitQuantity cannot be more than {UnitQuantity.MaxQty}"
     
-    let uQty = UnitQuantity.Create 50
-    match uQty with
-    | Ok (UnitQuantity qty) -> printfn $"Good: {qty}"
-    | Error errStr -> printfn $"Bad: {errStr}"
-
-Example of a class with a `private` constructor:
-
-    type MyClass private (count:int) =
-        member this.Count = count
-        static member CreateInstance (cnt: int) : MyClass option =
-            if cnt > 0
-            then Some(new MyClass(cnt))
-            else None
-
-    let myClass = MyClass.CreateInstance (5)
+    module UnitQuantity =  // common idiom: type companion module
+        let tryCreate qty =
+            if qty < 1 || qty > 100
+            then None
+            else Some (UnitQuantity qty)
+        let value (UnitQuantity uQty) = uQty
+        let zero = UnitQuantity 0
+    ...
+    let unitQtyOpt = UnitQuantity.tryCreate 5
+
+    let validQty =
+        unitQtyOpt
+        |> Option.defaultValue UnitQuantity.zero
 
 ### Recursive Reference
 
-F#'s type inference and name resolution runs in file and line order; by default, any forward references are considered as errors.
+F#'s type inference and name resolution runs in file and line order. By default, any forward references are considered errors.
```

Comment:
also left to right, e.g. the `(int, values) ||> X.fold (fun ...)` type inference thing mght be worth a mention somewhere
obv this is nice and focusd so not here